test/unit/services/HandTrackingService/HandTrackingService.spec.ts

```diff
@@ -0,0 +1,81 @@
+import * as sinon from 'sinon';
+import { afterEach, beforeEach, describe, it } from 'mocha';
+import { CanNotPerformPredictionError } from '../../../../src/errors/CanNotPerformPredictionError';
+import { expect } from 'chai';
+import { Hands } from '@mediapipe/hands';
+import { HANDS_INITIALIZE_STUB } from '../../../stubs/MEDIA_PIPE_HANDS.stub';
+import { HandTrackingService } from '../../../../src/services/HandTrackingService/HandTrackingService';
+import { IHandTrackingService } from '../../../../src/services/HandTrackingService/HandTrackingService.type';
+import { ServiceUnavailableError } from '../../../../src/errors/ServiceUnavailableError';
+import { SinonSpy } from 'sinon';
+
+describe(HandTrackingService.name, () => {
+  let handTrackingService: IHandTrackingService;
+  beforeEach(() => {
+    handTrackingService = new HandTrackingService();
+  });
+  afterEach(() => {
+    HANDS_INITIALIZE_STUB.resetHistory();
+  });
+  describe(HandTrackingService.prototype.initialize.name, () => {
+    it('should call hands initialize if app was NOT initialized', async () => {
+      await handTrackingService.initialize();
+      expect(HANDS_INITIALIZE_STUB).to.have.been.calledOnce;
+    });
+    it('should NOT call hands initialize 2nd if app was initialized 2 times', async () => {
+      await handTrackingService.initialize();
+      await handTrackingService.initialize();
+      expect(HANDS_INITIALIZE_STUB).to.have.been.calledOnce;
+    });
+  });
+  describe(HandTrackingService.prototype.start.name, () => {
+    let initializeSpy: SinonSpy;
+    beforeEach(() => {
+      initializeSpy = sinon.spy(handTrackingService, 'initialize');
+    });
+    afterEach(() => {
+      initializeSpy.restore();
```

Comment:
Why are you calling initializeSpy.restore(); in every test? Shouldn't this afterEach(); callback take care of it? If not what's the purpose of this beforeEach

Reply:
yes good catch thx